ocaml/xapi-cli-server/cli_operations.ml

```diff
@@ -5513,6 +5492,32 @@ let download_file rpc session_id task fd filename uri label =
       (* should never happen *)
       raise (ExitWithError 1)
 
+let download_file rpc session_id task fd filename uri label =
+  marshal fd (Command (HttpGet (filename, uri))) ;
+  let response = ref (Response Wait) in
+  while !response = Response Wait do
+    response := unmarshal fd
+  done ;
+  let ok =
+    match (!response, filename <> "") with
+    | Response OK, true ->
+        true
+    | Response OK, false ->
+        false
+    | Response Failed, _ ->
+        (* Need to check whether the thin cli managed to contact the server
+           				   or not. If not, we need to mark the task as failed *)
+        if Client.Task.get_progress ~rpc ~session_id ~self:task < 0.0 then
```

Comment:
Is `<` correct? Can this be negative? If not, it should be `<=`

Reply:
It probably can be negative
```ocaml
(* Initially mark the task progress as -1.0. The first thing the HTTP handler does it to mark it as zero *)
  (* This is used as a flag to show that the 'ownership' of the task has been passed to the handler, and it's *)
  (* not our responsibility any more to mark the task as completed/failed/etc. *)
  Client.Task.set_progress ~rpc ~session_id ~self:task ~value:(-1.0) ;
 ```

Reply:
So here the error category of "< 0.0" is that CLI has not contacted the server.